### PR TITLE
Modernizing IO a bit

### PR DESCRIPTION
**Pull request by philbucher**
**📝 Description**
Properly defaulting and deleting constructors

### CODE REVIEW

**Review by loumalouomega — COMMENTED** [review]
Okay from my side, wait for @KratosMultiphysics/technical-committee

File: kratos/includes/io.h [hunk 1]
```diff
@@ -108,16 +108,20 @@ class KRATOS_API(KRATOS_CORE) IO
...
     /// Default constructor.
-    IO() {}
+    IO() = default;
 
     /// Destructor.
-    virtual ~IO() {}
+    virtual ~IO() = default;
 
+    /// Copy constructor.
+    IO(IO const& rOther) = delete;
```
Comment (philbucher) [inline kratos/includes/io.h:117]: FYI deleting it in the baseclass propagates to derived classes (unless they define their own)

**Review by rubenzorrilla — APPROVED** [review]
Thanks.

File: kratos/includes/io.h (hunk 1 above)
Comment (rubenzorrilla) [reply kratos/includes/io.h:117]: Good to know.

**Pull request merged**